src/main/java/com/yoga/horus/controller/BaseController.java

```diff
@@ -1,18 +1,17 @@
 package com.yoga.horus.controller;
 
 import com.yoga.horus.util.APIResponse;
-import jakarta.servlet.http.HttpServletRequest;
-
+import java.util.List;
 import java.util.UUID;
 
 public interface BaseController<T> {
     APIResponse<T> getById(UUID id);
 
-    APIResponse<T> getAll();
+    APIResponse<List<T>> getAll();
 
-    APIResponse<T> create(HttpServletRequest request);
+    APIResponse<T> create(T object);
 
-    APIResponse<T> update(UUID id, HttpServletRequest request);
+    APIResponse<T> update(UUID id, T object);
 
-    APIResponse<T> delete(UUID id);
+    APIResponse<String> delete(UUID id);
 }
```

Comment:
We can modify it if all our controllers share the same functionalities "Basic CRUD"

No why should it be a bean ? the controllers actual implementation are the ones that should be beans

Reply:
After trying it though .. I think it is not a good idea to use a base controller like this because clearly not all controllers will have the same structure 
But if we really will have the same structure in them all then add Entity,DTO generics as well in the ``BaseController`` 
```java     
APIResponse<T> create(T object);
```
```java     
APIResponse<DTO> create(Entity object);
```
as we are not always getting/returning T 
we are returning a DTO and getting an Entity for now 
**NOTE: in case we do use DTOs for both request and response we can still handle this using generics for two types of DTOs --> RequestDTO and ResponseDTO**

Reply:
I think we should either close this PR until this discussion is done, or accept the PR and then work on this Refactor process.

I think we should accept it since all functionlities that are required in user controller are currently working as intended, and then have another branch for the refactor of Controllers.

@Mohamedsabry109 @MohanadKh03 @Ahmad-AlDeeb @Mstafa45 @mustafa-abuelmagd